### ISSUE TITLE
Relative path to local lambda folder

### ISSUE THREAD

**Issue by mikicho** [localstack/localstack#2376]
# Type of request: This is a ...

[ ] bug report
[x] feature request

# Detailed description
TL;DR: 
Does localstack expose the host PWD?
Or Does it possible to set relative path in lambda local `S3Key`

I'm working on a project and share code with my colleagues, we have different absolute paths to our project root.
We are trying to integrate lambda to our local stack but the `S3Key` requires an absolute path:
```
awslocal lambda create-function \
  --function-name myLambda \
  --code S3Bucket="__local__",S3Key="/path/to/folder" \
  --handler index.handler \
  --runtime nodejs10.x \
  --role whatever \
  --timeout 30 \
  --memory-size 512
```

But we can't commit it to git like that because it will work only on one machine.

We tried to set the folder as volume to localstack but from some reason we get:
```
localstack_1  | 2020-04-30T20:27:55:WARNING:localstack.services.awslambda.lambda_api: Error executing Lambda function arn:aws:lambda:us-west-2:000000000000:function:myLambda: Lambda process returned error status code: 1. Result: {"errorType":"Runtime.HandlerNotFound","errorMessage":"index.handler is undefined or not exported"}. Output:
localstack_1  | 2020-04-30T20:27:54.686Z	undefined	ERROR	Uncaught Exception	{"errorType":"Runtime.HandlerNotFound","errorMessage":"index.handler is undefined or not exported","stack":["Runtime.HandlerNotFound: index.handler is undefined or not exported","    at Object.module.exports.load (/var/runtime/UserFunction.js:144:11)","    at Object.<anonymous> (/var/runtime/index.js:45:30)","    at Module._compile (internal/modules/cjs/loader.js:778:30)","    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)","    at Module.load (internal/modules/cjs/loader.js:653:32)","    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)","    at Function.Module._load (internal/modules/cjs/loader.js:585:3)","    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)","    at startup (internal/bootstrap/node.js:283:19)","    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)"]}
```
Maybe its because docker can't (?) to create volume from volume folder?

Does localstack expose the host PWD?

Thanks
## Expected behavior
Localstack will mount the folder according to relative path from localstack's PWD

## Actual behavior
failed on ~~ `includes invalid characters for a local volume name,. If you intended to pass a host directory, use absolute path`

# Steps to reproduce
```
awslocal lambda create-function \
  --function-name myLambda \
  --code S3Bucket="__local__",S3Key="./relative/path/to/folder" \
  --handler index.handler \
  --runtime nodejs10.x \
  --role whatever \
  --timeout 30 \
  --memory-size 512
```

## Command used to start LocalStack
docker compose

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
awslocal

**Comment by mikicho** [comment]
A workaround (or a solution?) can be set`ROOT` env var:
```
  localstack:
    image: localstack/localstack
    environment:
      - SERVICES=sqs,lambda
      - DEBUG=1
      - DEFAULT_REGION=us-west-2
      - LAMBDA_REMOTE_DOCKER=false
      - ROOT=$PWD
    volumes:
      - "$PWD/localstack:/docker-entrypoint-initaws.d"
    env_file:
      - .env
```

```
awslocal lambda create-function \
  --function-name myLambda \
  --code S3Bucket="__local__",S3Key="$ROOT/path/to/folder" \
  --handler index.handler \
  --runtime nodejs10.x \
  --role whatever \
  --timeout 30 \
  --memory-size 512
```

**Comment by whummer** [comment]
Thanks for reporting @mikicho , and thanks for posting a possible solution. I think you could also use `$PWD` directly, and remove $ROOT from your configuration, right?
```
...
--code S3Bucket="__local__",S3Key="$PWD/path/to/folder" \
...
```
Depending on how your project setup looks like, there should be different options (based on $PWD) on how to achieve this behavior, so you can safely share your code with your colleagues. Let us know if there are any other issues. Thanks

**Comment by mikicho** [comment]
@whummer thanks
`$PWD` is working for my local environment.
It fails on Jenkins for some reason.
Does localstack auto-volume the `pwd`? maybe on Jenkins it fails silently on permissions problem?
